src/test/java/org/zalando/nakadi/controller/SubscriptionControllerTest.java

```diff
@@ -176,8 +194,33 @@ public void whenGetNoneExistingSubscriptionThenNotFound() throws Exception {
                 .andExpect(content().string(jsonHelper.matchesObject(expectedProblem)));
     }
 
+    @Test
+    public void whenPostSubscriptionThenWithNoReadScope() throws Exception {
+        when(eventTypeRepository.findByName("myET")).thenReturn(EventTypeTestBuilder.builder()
+                .name("myET")
+                .readScopes(Collections.singleton("oauth.read.scope"))
+                .build());
+
+        expectedException.expectCause(IsInstanceOf.instanceOf(IllegalScopeException.class));
+        expectedException.expectMessage("Client has to have scopes: [oauth.read.scope]");
```

Comment:
We do not get proper exception handling for exception which are in ExceptionHandling.java,  because we do not start the whole spring application and ExceptionHandling class is not initialized. There are ways to add this to current mvc mock but it looks clumsy for me.